Revamp styling of the timeline; COUNTRY=rbd

JulienPathtech opened the pull request:
This fixes issue #787.

This complete the PR https://github.com/WFP-VAM/prism-app/pull/852 by updating style according to sketch

How to test the feature:

- [x] Use rbd with 2 layers => https://prism-882.surge.sh/?baselineLayerId=ch_phase&date=2023-05-27&hazardLayerIds=rainfall_dekad
- [x] Use southsudan with 3 layers => http://localhost:3000/?hazardLayerIds=flood_events%2Crainfall_dekad%2Cconflict_fatalities&date=2022-12-11
- [x] Move cursor of the timeline
- [x] Select date from the date picker or the timeline

Screenshot/video of feature:
![image](https://github.com/WFP-VAM/prism-app/assets/100351051/604d8399-4637-4068-84d1-c3936a5c4f75)


Figma designs:
![image](https://github.com/WFP-VAM/prism-app/assets/16843267/c16a9b8f-5ab2-4967-9e60-0cc0750807f7)

JulienPathtech commented:
There is a known bug when we try to select any date on the Timeline. I'm on it :)

wadhwamatic commented:
> Hi @wadhwamatic, I kept the grey layer that represent available or selectable dates in the calendar. Should we keep that layer or not ?

@JulienPathtech - I don't see value in the grey layer anymore since the intersections are clearly illustrated by the individual layer's bars. Selectable dates in the calendar should still appear though and that is based on the intersection of selected layers. Hope this is clear!

wadhwamatic commented:
@JulienPathtech - great start on this! I took a quick look for now and noticed a bug when you toggle between English and French (click 'FR') in the top right

Minor graphical issue: I expected the rectangle in the timeline and the handle for the slider to move together. I see that the rectangle waits until the layers are loaded. Instead, perhaps the rectangle should disappear until the layers are loaded for the new date.

JulienPathtech commented:
**Update**
- Fix when selected a point in the Timeline (https://github.com/WFP-VAM/prism-app/pull/882#issuecomment-1599108150)
- Removed grey / intersection layer
- Make the whole cursor moving when selected (There is a drawback to this -> tooltip is not available arround the cursor : /. If I use a lower z-index on the cursor, it breaks its display). We can keep it or revert to previous version

I'm waiting your feedbacks

ericboucher commented:
Looks great. I think it's ok this way @JulienPathtech, it only blocks the tooltip and clicks for the day before and the day after, which I think is fine since there are two other ways to move to these dates already.

ericboucher commented on frontend/src/components/MapView/DateSelector/index.tsx:
```diff
@@ -335,10 +325,11 @@ const DateSelector = memo(
             <Hidden smUp>
               <Button onClick={decrementDate}>
-                <ChevronLeft />
+                <ChevronLeft style={{ color: '#101010' }} />
```
It might be worth extracting some of these colors into the theme?

JulienPathtech replied:
How do I do that without possibly breaking other component current color ? 
I could add a `color.secondary` item with that color but it would not reflec actual application theme

ericboucher commented on frontend/src/types/custom.d.ts:
```diff
@@ -3,6 +3,8 @@
 // Allow svg imports in .tsx
 // https://stackoverflow.com/questions/44717164/unable-to-import-svg-files-in-typescript
 declare module '*.svg' {
+  import React = require('react');
+  export const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
```
are these needed?

JulienPathtech replied:
Yep, to enable importing SVG element at compilation time :)

ericboucher reviewed (approved):
Great work @JulienPathtech!
Let's take this opportunity to add a bit more comments around the timeline items maybe?

HarryMytilinaios commented:
@JulienPathtech great work this feels really nice I agree with the comments of @ericboucher regarding the colors in the theme, if that is possible, also another addition seems to me that the dateSelector component is a little bit higher, would be nice if it was a little bit lower to give more space in the map because with the legends enabled this seems a little bit too crowded wdyt @ericboucher @wadhwamatic ?

wadhwamatic commented:
@JulienPathtech - this is looking good but I still see an issue when I toggle to French - the cursor moves to 1-January. 

English:
<img width="1512" alt="Screenshot 2023-06-22 at 22 14 02" src="https://github.com/WFP-VAM/prism-app/assets/3343536/eab4d43d-4624-4a4a-b2ad-5057ec110109">

French:
<img width="1512" alt="Screenshot 2023-06-22 at 22 14 20" src="https://github.com/WFP-VAM/prism-app/assets/3343536/6eb30296-df69-4b12-9986-4e8261fdfd06">

ericboucher commented:
Looks like this bug was introduced with the UX revamp. @JulienPathtech can you have a look? My guess would be that this has something to do with the fact that French dates and English dates have month and days inverted

ericboucher commented:
Update, I found the issue, which was actually due to the fact that Frenchies don't capitalize months! There might be a better way to do it than what I did @JulienPathtech. Since you've been head down in this component for a few days, let me know if you can think of a better approach.

JulienPathtech commented:
Update : 
- @ericboucher I use the `datesAreEqualWithoutTime` helper. 
- Added comments 

For your point about the Timeline @HarryMytilinaios , I agree. Here are my problems : 
- If I lower / center the Timeline the SVG desapear at the bottom
- I reworked the SVG cursor so it perfectly match the Timeline in size but it's height and width are fixed

So I don't find any good solution to lower the Timeline that ATM.

ericboucher commented on frontend/src/components/MapView/DateSelector/index.tsx:
```diff
@@ -149,16 +149,10 @@ const DateSelector = memo(
-      });
-    }, [locale, range, stateStartDate]);
+      return findIndex(range, date =>
+        datesAreEqualWithoutTime(date.value, stateStartDate || 0),
```
how about `stateStartDate && datesAreEqualWithoutTime(date.value, stateStartDate)` instead? Same result but maybe a bit more straightforward?

ericboucher commented on frontend/src/components/MapView/DateSelector/TimelineItems/index.tsx:
```diff
@@ -111,8 +105,9 @@ const TimelineItems = memo(
-    }, [selectedLayerDateItems, timelineStartDate]);
+    }, [selectedLayers, timelineStartDate]);
 
+    // Draw a colomn for each date of the Timeline that start at the begining of the year
```
column?
